src/Pagination/PaginationManipulator.php

```diff
@@ -47,15 +47,15 @@ public static function transformToPaginatedField(
      * @param  \Nuwave\Lighthouse\Schema\AST\DocumentAST  $documentAST
      * @param  int|null  $defaultCount
      * @param  int|null  $maxCount
-     * @return \Nuwave\Lighthouse\Schema\AST\DocumentAST
+     * @return void
      */
     public static function registerConnection(
-        FieldDefinitionNode $fieldDefinition,
-        ObjectTypeDefinitionNode $parentType,
-        DocumentAST $documentAST,
+        FieldDefinitionNode &$fieldDefinition,
+        ObjectTypeDefinitionNode &$parentType,
+        DocumentAST &$documentAST,
```

Comment:
Why is is needed to pass them by reference?

Reply:
If we need to add custom types, they are added to `$documentAST->types`, which makes them available globally. I think that is why it is added with `&`.

Reply:
Trying to shoehorn a non-mutable style into the AST manipulation has caused lots of additional complexity and issues such as #636 

It is not really needed to pass them by reference, since objects are mutable anyway, but i think it is more explicit and obvious that way.

Reply:
I agree with keeping it mutable. Was just wondering if there were any reason for it, but I guess by reference is faster when we are dealing with an AST, as they are often pretty big. 
Not that the speed impact for that should matter :)